### DISCUSSION TITLE
Troubleshooting ECONNREFUSED Error When Connecting Flowise and LocalAI

### DISCUSSION THREAD

**Discussion by nattyraz** [FlowiseAI/Flowise#304] [Q&A] [answered]
Hello,

I've been working on setting up Flowise and LocalAI locally on my machine using Docker. Although I'm not an expert in coding, I've managed to get some systems running locally. I've ensured the appropriate model files are in the /models directory and the setup for Pinecone and OpenAI API is correct.

Both Flowise and LocalAI are accessible at their respective addresses: http://localhost:3000/ for Flowise and http://localhost:8080/ for LocalAI. However, I'm encountering an issue when trying to connect to the LocalAI server from the ChatLocalAI module in the Conversational Retrieval QA Chain on Flowise. Specifically, I'm receiving an ECONNREFUSED error on 127.0.0.1:8080 when I send a request.

To troubleshoot, I've tested this setup on two different PCs, but I'm consistently encountering the same error. Interestingly, I was able to communicate with a model directly in VSCode, but I've been unsuccessful in doing so within Flowise.

Any guidance on how to resolve this issue would be greatly appreciated.

**Comment by HenryHengZJ** [comment, marked as answer]
if you are running both flowise and localai on docker, give this a try - https://github.com/go-skynet/LocalAI/tree/master/examples/flowise

docs - https://docs.flowiseai.com/chat-models/chatlocalai

**Reply by nattyraz** [reply]
yooooo !!! that good man !!

**Comment by hannahnguyen14** [comment]
Hi, I got quite the same issue too, do you know how to fix it yet? The setup for my flowise is correct, and the response from curl http://localhost:8080/v1/models is successful. However, when I go to http://localhost:8080/, it returns errors like this: {"error":{"code":404,"message":"Cannot GET /","type":""}}. I'm not sure if it's because my LocalAI is still not accessible. 

Do you have any guidance from your experience?
